### DISCUSSION TITLE
Интеграция  ldap  google   авторизации

### DISCUSSION THREAD

**Discussion by smplseven** [totumonline/totum-ru-issues-and-discussions#346] [ОШБК!] [answered]
Подключили и установили настройки  ldap  авторизации ,импортировали пользователей.
Но возникает такая проблема ,что когда пытаемся авторизоваться выходит ошибка (см.на скрине ) и отключается пользователь в поле on_off  общей таблицы Пользователей. Если вручную включить пользователя и обновить страницу где возникала ошибка, пользователь показывается авторизованным. Есть возможность подсказать - куда смотреть?
<img width="1310" alt="Снимок экрана 2023-04-06 в 13 59 46" src="https://user-images.githubusercontent.com/113604604/230358045-78beb4a3-f0a9-4216-81d5-2dfc06c8c3c7.png">

**Comment by ttm-alexeypolunin** [comment]
Посмотрим сег. Что у вас является сервером LDAP? Тк мы тестировали на AD и несмотря на то, что пытались предусмотреть все варианты, что-то пошло не так :)

**Reply by smplseven** [reply]
google

**Comment by poluninatm** [comment]
Отключение пользователя происходит в h_import_users таблицы настройки LDAP после проверки его на сервере LDAP на активность и соответствие требованиям настроек.

Очень похоже, что у вас происходит ситуация, на которую мы не рассчитывали - биндинг пользователя проходит, но пользователь не возвращается при проверке его настроек на сервере LDAP (или не получает нужных ролей?) - и отключается в тотуме - поэтому вырастает каскад ошибок - авторизация прошла, но данные по пользователю система не получает потому что он в этот момент уже выключается.

**Reply by poluninatm** [reply]
Чтобы посмотреть, как это происходит, можно воспользоваться кнопкой "h_update_one_user"

**Comment by poluninatm** [comment]
"Если вручную включить пользователя и обновить страницу где возникала ошибка, пользователь показывается авторизованным" - потому что к моменту, когда выпадает fatal в сессию уже прописывается пользователь. Если он выключается LDAP проверкой - он теряет авторизацию на следующей загрузке страницы, если вы его включаете вручную - он оказывается прошедшим авторизацию и включенным.

**Comment by poluninatm** [comment, marked as answer]
Одиночный пользователь с сервера LDAP получается в строке `ldapSelectSingle:` кода кнопки `h_import_users`. Этот запрос использует настройку `filter_single` конкретного домена авторизующегося пользователя (`#h_domains_settings[$#domain][filter_single]`) из поля `h_domains_settings` с подстановкой параметров `{USER_LOGIN_PARAM}` и `{USER_LOGIN}`. 

Проверить работоспособность этой конструкции можно кнопкой `h_update_one_user` (`Обновить одного пользователя`)

**Reply by smplseven** [reply]
Когда мы проверяем юзера по кнопке (h_update_one_user) у нас выходит окно с данными по юзеру  ,что с ними делать и какие параметры  нам нужно проверить и  если что потом исправить, могли подсказать пжл?

**Reply by ttm-alexeypolunin** [reply]
напишите мне в телеграм `@alexeypolunin` тк у нас нет тесткейса с google мы вам дистанционно не поможем — только подключившись и посмотрев
